tests/mockUtil.ts

```diff
@@ -1,11 +1,58 @@
 import { resolve } from "path";
-import { OutputChannel, Uri } from "vscode";
+import {
+  DiagnosticCollection,
+  FileSystemWatcher,
+  OutputChannel,
+  Uri,
+} from "vscode";
 import * as Module from "module";
 
 const originalRequire = Module.prototype.require;
 const oRegisteredCommands: Record<string, unknown> = {};
 const outputChannel = { show: () => "", append: () => "" };
 
+export interface Disposable {
+  dispose(): unknown;
+}
+
+class MockPosition {
+  constructor(public line: number, public character: number) {}
+}
+
+class MockRange {
+  constructor(public start: MockPosition, public end: MockPosition) {}
+}
+
+export const mockFileSystemWatcher: FileSystemWatcher = {
+  ignoreChangeEvents: false,
+  ignoreCreateEvents: false,
+  ignoreDeleteEvents: false,
+  onDidChange: (): Disposable => {
```

Comment:
Optional - if you want to make it shorter you can use
```typescript
onDidChange: (): Disposable => ({ dispose: () => undefined })
```
(For all the events in this file)

Reply:
done